frontend/packages/react-app/src/components/pages/ExplorePage.tsx

```diff
@@ -1,17 +1,53 @@
-import React, {useCallback, useEffect, useState} from "react"
-import { Web3Provider } from "@ethersproject/providers"
-import { web3Modal } from "../../utils/web3Modal"
-import ExplorePageTemplate from "../templates/ExplorePageTemplate"
-import { GET_TOKENS_BALANCE_USER_HOLDS } from "../../graphql/subgraph"
-import { useLazyQuery } from "@apollo/react-hooks"
-import { ethers } from "ethers"
-import {TokenBalanceUserHolds, UserToken} from "../../interfaces";
+import React, { useCallback, useEffect, useState, useReducer } from "react";
+import { Web3Provider } from "@ethersproject/providers";
+import { web3Modal } from "../../utils/web3Modal";
+import ExplorePageTemplate from "../templates/ExplorePageTemplate";
+import { GET_TOKENS_BALANCE_USER_HOLDS } from "../../graphql/subgraph";
+import { useLazyQuery } from "@apollo/react-hooks";
+import { ethers } from "ethers";
+import { TokenBalanceUserHolds, UserToken } from "../../interfaces";
+
+export const ACTIONS = {
+  OPEN_MODAL: "open_modal",
+  CLOSE_MODAL: "close_modal",
+  SET_TOKENS: "set_tokens",
+};
+
+export interface ExplorePageState {
+  isOpen: boolean;
+  tokens: UserToken[];
+}
+
+interface ExplorePageAction {
+  type: string;
+  payload: {
+    tokens: UserToken[];
+  };
+}
+
+function reducer(state: ExplorePageState, action: ExplorePageAction) {
+  switch (action.type) {
+    case ACTIONS.OPEN_MODAL:
+      return { ...state, isOpen: true };
+    case ACTIONS.CLOSE_MODAL:
+      return { ...state, isOpen: false };
+    case ACTIONS.SET_TOKENS:
+      return { ...state, tokens: action.payload.tokens };
+    default:
+      throw new Error();
+  }
+}
+
+const initialState = { isOpen: false, tokens: [] };
 
 const ExplorePage = () => {
-  const [provider, setProvider] = useState<Web3Provider>()
-  const [tokens, setTokens] = useState<UserToken[]>([])
-  const [walletAddress, setWalletAddress] = useState("")
-  const [getTokensBalance, { loading, error, data }] = useLazyQuery<TokenBalanceUserHolds>(GET_TOKENS_BALANCE_USER_HOLDS)
+  const [provider, setProvider] = useState<Web3Provider>();
+  const [walletAddress, setWalletAddress] = useState("");
```

Comment:
Will this states are moved into state? Just work in progress?

Reply:
We haven't done anything here yet at this point.
I think it would be appropriate to manage it as a global state, but I don't understand it yet.